autopush/websocket.py

```diff
@@ -270,21 +270,26 @@ def process_hello(self, data):
         self.uaid = uaid
 
         connect = data.get("connect")
-        if connect and self.ap_settings.pinger:
+        if connect and self.ap_settings.bridge:
             self.transport.pauseProducing()
-            d = deferToThread(self.ap_settings.pinger.register, uaid, connect)
+            d = deferToThread(self.ap_settings.bridge.register, uaid, connect)
             d.addCallback(self._check_router, True)
             d.addErrback(self.err_hello)
         else:
             self._check_router(False)
 
-    def _check_router(self, paused=False):
+    def _check_router(self, paused=False, bridge_register=None):
         if paused:
             self.transport.resumeProducing()
+        if bridge_register is not None and bridge_register is False:
```

Comment:
> I think that registration failure needs to be communicated back to the client, but that's also why I don't terminate the socket connection on a bridge failure.

Agreed. Should we remove the `return` on line 289 and let the normal handshake take over, then, or wait for the client to retry?